src/Microsoft.Graph/Models/Extensions/DateTimeZoneExtensions.cs

```diff
@@ -0,0 +1,181 @@
+
+        /// <summary>
+        /// Converts a Datetime parameter to its equivalent DateTimeTimeZone Complex Type given its TimeZone
+        /// </summary>
+        /// <param name="dateTime"></param>
+        /// <param name="timeZone"></param>
+        /// <returns></returns>
+        public static DateTimeTimeZone FromDateTime(DateTime dateTime, string timeZone)
+        {
+            TimeZoneInfo timeZoneInfo = GetTimeZoneInfo(timeZone);
+
+            // When in Local or Utc kind, we can convert to the time zone specified
+            // Which may be a no-op if the value is already in the specified time zone
+            // We don't do this for Unspecified kind, as that would infer converting *from* the local time zone
+            if (dateTime.Kind != DateTimeKind.Unspecified)
+            {
+                dateTime = TimeZoneInfo.ConvertTime(dateTime, timeZoneInfo);
+            }
+
+            DateTimeTimeZone dateTimeTimeZone = new DateTimeTimeZone
+            {
+                DateTime = dateTime.ToString(DateTimeFormat, CultureInfo.InvariantCulture),
+                TimeZone = timeZone
+            };
+
+            return dateTimeTimeZone;
+        }
+
+        /// <summary>
+        /// Converts a DatetimeOffset parameter to its equivalent DateTimeTimeZone Complex Type given its TimeZone
+        /// </summary>
+        /// <param name="dateTimeOffset"></param>
+        /// <param name="timeZone"></param>
+        /// <returns></returns>
+        public static DateTimeTimeZone FromDateTimeOffset(DateTimeOffset dateTimeOffset, string timeZone)
+        {
+            TimeZoneInfo timeZoneInfo = GetTimeZoneInfo(timeZone);
+
+            // Convert the input value to the time zone specified.
+            // If it's already in an offset valid for that time zone, this will be a no-op.
+            dateTimeOffset = TimeZoneInfo.ConvertTime(dateTimeOffset, timeZoneInfo);
+
+            DateTimeTimeZone dateTimeTimeZone = new DateTimeTimeZone
+            {
+                DateTime = dateTimeOffset.ToString(DateTimeFormat, CultureInfo.InvariantCulture),
+                TimeZone = timeZone
+            };
+
+            return dateTimeTimeZone;
+        }
+
+        internal static TimeZoneInfo GetTimeZoneInfo(string timeZone)
+        {
+            // The following is built-in, but is platform dependent.
+            // It will use Windows time zones on Windows, and IANA time zones on other platforms.
+
+            return TimeZoneInfo.FindSystemTimeZoneById(timeZone);
+
+
+            // The following is platform independent, working with either Windows or IANA time zones on any platform.
+            // However, it requires a dependency on https://github.com/mj1856/TimeZoneConverter
+            // See also https://github.com/dotnet/corefx/issues/11897
+            //      and https://github.com/dotnet/corefx/issues/2538
+            //
+            // return TimeZoneConverter.TZConvert.GetTimeZoneInfo(timeZone);
```

Comment:
Would be good to make a determination on this.  If acceptable to take a dependency, then using TimeZoneConverter would more closely align to the Graph API's idea of time zones.

Reply:
Matt, thank you for making us aware of some of complexities around time zones. Handling timezones correctly is currently outside the scope of our clients. We need the service workloads to drive conformity and the time zone standards for clients.  We won't be doing any time zone conversion.